src/android/PGMultiView.java

```diff
@@ -34,31 +72,37 @@
      * @param args JSONArry of arguments for the plugin.
      * @param callbackContext the callbackContext used when calling back into JavaScript.
      * @return boolean success or fail
-     */
-    public boolean execute(String action, CordovaArgs args, final CallbackContext callbackContext) throws JSONException {
-        if (action.equals("loadView")) {
-            final String url = args.getString(0);
 
-            Log.d(LOG_TAG, "url = " + url);
 
-            intent = new Intent(this.cordova.getActivity(), com.phonegap.PGMultiViewdemo.MainActivity.class);
-            // todo: figure out how to pass this in
-            //intent.putExtra("startUrl", "http://www.phonegap.com");
-            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
 
-            this.cordova.getActivity().startActivity(intent);
+    //cordova is context of main activity.  webView is the webview we are running the main Activity (cordova) inside of.
+     public void initialize(CordovaInterface cordova, CordovaWebView webView) {
+    super.initialize(cordova, webView);
+  }
+    public boolean execute(String action, CordovaArgs args, final CallbackContext callbackContext) throws JSONException {
+        if (action.equals("loadView") &&args.length() > 0) {
+            final String url = args.getString(0);
 
+            Log.d(LOG_TAG, "url = " + url);
+            openWebView(url);
             PluginResult pluginResult = new PluginResult(PluginResult.Status.OK);
             pluginResult.setKeepCallback(true);
             callbackContext.sendPluginResult(pluginResult);
-
-
         }
         else if (action.equals("dismissView")) {
-            // todo: more to it than this
-            this.cordova.getActivity().finish();
-        }
-        return true;
+            quit();
+
+    private void viewWebView(final String url){ //url is passed in from execute, its args.getstring(0)
```

Comment:
nitpick: `startCordovaActivity` would be a better name for this method, rather than viewWebView